### PR TITLE
Enable Additional Pre-commit Hooks and Fix Errors

### PR DESCRIPTION
**Pull request by danyeaw**
This PR enables the following pre-commit hooks:

- flake8
- docformatter
- shellcheck

It also fixes errors found by as part of enabling the new checks. These will be run automatically along with Black in the lint phase of the CI build.

### CODE REVIEW

File: gvsbuild/projects.py [hunk 1]
```diff
@@ -51,11 +50,13 @@ def build(self):
 
         self.push_location(r".\win32")
         self.exec_vs(
-            r'nmake /nologo /f adwaita-msvc.mak CFG=%(configuration)s PYTHON="%(python_dir)s\python.exe" PREFIX="%(gtk_dir)s"',
```
Comment (nacho) [inline gvsbuild/projects.py:54]: can we increase the line size?

Comment (danyeaw) [reply gvsbuild/projects.py:54]: While it is possible to increase the line length, I think we should just stick to the Black defaults. Here is the [full description](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length) of why line length is set to 88, but the tldr of it is that it is trying to get close to PEP8 of 79, with some extra wiggle room to reduce file length. The positive is that Black does a great job of wrapping lines when they are too long.

Comment (nacho) [reply gvsbuild/projects.py:54]: while I normally I agree about it, this is wrapping even tarball links which is just bad... I wonder if we can specify that strings should not be wrapped

Comment (danyeaw) [reply gvsbuild/projects.py]: @nacho I looked at this closer, and you were right - thanks for catching this! I was able to undo breaking up the raw strings while still allowing black and flake8 to be happy. 👍

File: build.py [hunk 2]
```diff
@@ -17,9 +17,13 @@
 
 """Main build script."""
 
+import gvsbuild.groups  # noqa: F401
+import gvsbuild.projects  # noqa: F401
+import gvsbuild.tools  # noqa: F401
```
Comment (pbor) [inline build.py:22]: this looks wrong from a cursory view: the lines below try to adjust the path before importing more things, no?

Comment (danyeaw) [reply build.py]: Hi @pbor, thanks for pointing this out. I am not sure why these imports got pulled up, but I moved them back down to where they were before.

**Comment by pbor** [comment]
great!

**Comment by nacho** [comment]
cool stuff

**Pull request merged**